subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/Waiter.kt

```diff
@@ -11,7 +11,7 @@ import androidx.test.espresso.ViewInteraction
 fun <T> waitFor(
     frequencyMs: Long = UITestConfig.waiterFrequencyMs,
     timeoutMs: Long = UITestConfig.waiterTimeoutMs,
-    allowedExceptions: Set<Class<out Throwable>> = UITestConfig.waiterAllowedExceptions,
+    allowedExceptions: Set<Class<out Any>> = UITestConfig.waiterAllowedExceptions,
```

Comment:
cause 

```
public interface EspressoException extends TestFrameworkException
public interface TestFrameworkException
```

Reply:
Maybe we cant extend our exception from that interface. It will make the contract more clear

Reply:
But we don't have the exception here, only marker interface (TestFrameworkException).

And I don't want to wrap all different errors in our single type. We had this approach partially before and it turned out as fragile. Even more, I don't see profits for the user to have our custom error from the test framework.